src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -131,8 +161,59 @@ class JesBackend extends Backend with LazyLogging {
   override def adjustInputPaths(call: Call, inputs: CallInputs): CallInputs = inputs map { case (k, v) => mapInputValue(k, v) }
   override def adjustOutputPaths(call: Call, outputs: CallOutputs): CallOutputs = outputs
 
-  // No need to copy GCS inputs for the workflow we should be able to directly reference them
-  override def initializeForWorkflow(workflow: WorkflowDescriptor): Try[HostInputs] = Success(workflow.actualInputs)
+  /*
+   * No need to copy GCS inputs for the workflow we should be able to directly reference them
+   * Create an authentication json file containing docker credentials and/or user account information
+   */
+  override def initializeForWorkflow(workflow: WorkflowDescriptor, dataAccess: DataAccess): Try[HostInputs] = {
+    val userJson = if (authenticationMode == RefreshTokenMode) {
+      storeGcsAuthInformation(workflow, dataAccess)
+    } else None
+
+    GcsAuth.generateJson(DockerHubCredentials, userJson) foreach { content =>
+      val path = GoogleCloudStoragePath(gcsAuthFilePath(workflow))
+      logger.info(s"Creating authentication file for workflow ${workflow.id} at \n ${path.toString}")
+      JesConnection.storage.uploadJson(path, content)
+    }
+
+    Success(workflow.actualInputs)
+  }
+
+  override def assertWorkflowOptions(options: Map[String, String]) = {
+    // Warn for unrecognized option keys
+    options.keySet.diff(OptionKeys) match {
+      case unknowns if unknowns.nonEmpty => logger.warn(s"Unrecognized workflow option(s): ${unknowns.mkString(", ")}")
+      case _ =>
+    }
+
+    if (authenticationMode == RefreshTokenMode) {
+      Seq(AccountOptionKey, RefreshTokenOptionKey) filterNot options.keySet match {
+        case missing if missing.nonEmpty =>
+          throw new IllegalArgumentException(s"Missing parameters in workflow options: ${missing.mkString(", ")}")
+        case _ =>
+      }
+    }
+  }
+
+  /**
+   * Delete authentication file in gcs once workflow is in a terminal state.
+   */
+  override def cleanUpForWorkflow(workflow: WorkflowDescriptor)(implicit ec: ExecutionContext) = {
+    Future(JesConnection.storage.deleteObject(GoogleCloudStoragePath(gcsAuthFilePath(workflow))))
+  }
+
+  //TODO: Store information in encrypted K/V Cromwell DB (hence the dataAccess parameter)
+  /**
+   * Store user auth information in encrypted cromwell DB and return a GcsUserAuthInformation
+   */
+  def storeGcsAuthInformation(workflow: WorkflowDescriptor, dataAccess: DataAccess) = {
+    val gcsAuthInfo = for {
+      account <- workflow.workflowOptions.get(AccountOptionKey)
+      token <- workflow.workflowOptions.get(RefreshTokenOptionKey)
+    } yield GcsUserAuthInformation(account, token)
+//    gcsAuthInfo foreach dataAccess.storeGcsAuthInfo // Not implemented yet
+    gcsAuthInfo
+  }
```

Comment:
Ok, this function actually don't do the storage currently but create a GcsUserAuthInformation to be passed to the function that generate the Json.